### ISSUE TITLE
Global minimum > Local minimum

### ISSUE THREAD

**Issue by daibo83** [komora-io/ebr#2]
When benchmarking multithreaded insert performance of sled 1.0.0-alpha.120, this error would occasionnally occur when inserts happens from more than 4 threads.

```
thread '<unnamed>' panicked at /home/daibo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ebr-0.2.11/src/lib.rs:333:21:
global minimum: 9292, ours: 9290
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at src/main.rs:100:23:
called `Result::unwrap()` on an `Err` value: Any { .. }
```

This is the benchmark code

```
fn sled_multi_threaded_put(items: usize, threads: usize) {
    let path = "./sled";
    let db = sled::open(path).unwrap();
    let mut handles = vec![];
    for t in 0..threads {
        let db = db.clone();
        handles.push(std::thread::spawn(move || {
            for i in (t * items / threads)..((t + 1) * items / threads) {
                let key = i.to_be_bytes();
                db.insert(key.as_ref(), key.as_ref()).unwrap();
            }
        }));
    }
    for handle in handles {
        handle.join().unwrap();
    }
}
```

**Comment by bojanserafimov** [comment]
+1 I stepped on this today using `ebr = 0.2.11` while evaluating `concurrent-map = 5.0.33` with default GC settings

**Comment by spacejam** [comment]
Thank you for reporting this! After a lot of head scratching and incorrect assumptions that my ARM laptop would always be a harsher test environment than my dusty old x86_64 laptop, I realized a barrier actually needed to be stronger on x86_64 to have it not be ignored at run time. [Fixed in version `0.2.13`](https://github.com/komora-io/ebr/commit/49007a120dce9305af61c7c16aa280f4194aca00)! This was an interesting bug where x86_64 seems to need a stronger memory ordering than ARM, at least on the systems I tested it on. I feel like that means that the fence in the pin function should have always been SeqCst for correctness, but it's the first time I've needed to make orderings stronger to get them to pass on x86_64 instead of ARM. [On x86_64 non-SeqCst orderings are just treated as compiler fences though](https://en.cppreference.com/w/cpp/atomic/atomic_thread_fence#Notes).
